mempool/src/error/ban_score.rs

```diff
@@ -480,6 +487,44 @@ impl MempoolBanScore for CheckTransactionError {
             CheckTransactionError::TxSizeTooLarge(_, _, _) => 100,
             CheckTransactionError::DeprecatedTokenOperationVersion(_, _) => 100,
             CheckTransactionError::HtlcsAreNotActivated => 100,
+            CheckTransactionError::OrdersAreNotActivated(_) => 100,
+            CheckTransactionError::OrdersCurrenciesMustBeDifferent(_) => 100,
+        }
+    }
+}
+
+impl MempoolBanScore for orders_accounting::Error {
+    fn mempool_ban_score(&self) -> u32 {
+        use orders_accounting::Error;
+
+        match self {
+            Error::StorageError(_) => 0,
+            Error::AccountingError(_) => 100,
+            Error::OrderAlreadyExists(_) => 100,
+            Error::OrderDataNotFound(_) => 0,
+            Error::OrderAskBalanceNotFound(_) => 0,
+            Error::OrderGiveBalanceNotFound(_) => 0,
+            Error::OrderWithZeroValue(_) => 100,
+            Error::InvariantOrderDataNotFoundForUndo(_) => 100,
+            Error::InvariantOrderAskBalanceNotFoundForUndo(_) => 100,
+            Error::InvariantOrderAskBalanceChangedForUndo(_) => 100,
+            Error::InvariantOrderGiveBalanceNotFoundForUndo(_) => 100,
+            Error::InvariantOrderGiveBalanceChangedForUndo(_) => 100,
+            Error::InvariantOrderDataExistForCancelUndo(_) => 100,
+            Error::InvariantOrderAskBalanceExistForCancelUndo(_) => 100,
+            Error::InvariantOrderGiveBalanceExistForCancelUndo(_) => 100,
+            Error::InvariantNonzeroAskBalanceForMissingOrder(_) => 100,
+            Error::InvariantNonzeroGiveBalanceForMissingOrder(_) => 100,
```

Comment:
Not that it's a big deal, but IMO it's better for "invariant"  errors to have zero ban score, because they can't be peer's fault.
Same for `chainstate/src/detail/ban_score.rs`

Reply:
Maybe, not sure, it's just how it's done for every other invariant error